vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go

```diff
@@ -113,10 +114,10 @@ func (s *externalMetricsClient) List(ctx context.Context, namespace string, opts
 					return nil, err
 				}
 				if m == nil || len(m.Items) == 0 {
-					klog.V(4).Infof("External Metrics Query for VPA %+v: resource %+v, metric %+v, No items,", vpa.ID, resourceName, metricName)
+					klog.V(4).Infof("External Metrics Query for VPA %s: resource %+v, metric %+v, No items,", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName), resourceName, metricName)
```

Comment:
Just a sanity check as I can't dive deeper on my current device. Can ID be nil?

/hold

Reply:
In https://github.com/kubernetes/autoscaler/blob/628fc39443098c7e6683da739c62accf39100c51/vertical-pod-autoscaler/pkg/recommender/model/cluster.go#L248 it seems that vpaID is always set to Vpa model.

If `vpa.ID` is nil, it would panic earlier at https://github.com/kubernetes/autoscaler/blob/1739a71d80acba2f0ccacd741fd44b747c2b407b/vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go#L105.

/hold cancel